lib/core.ps1

```diff
@@ -839,18 +839,28 @@ function is_scoop_outdated() {
 }
 
 function substitute($entity, [Hashtable] $params, [Bool]$regexEscape = $false) {
-    if ($entity -is [Array]) {
-        return $entity | ForEach-Object { substitute $_ $params $regexEscape}
-    } elseif ($entity -is [String]) {
-        $params.GetEnumerator() | ForEach-Object {
-            if($regexEscape -eq $false -or $null -eq $_.Value) {
-                $entity = $entity.Replace($_.Name, $_.Value)
-            } else {
-                $entity = $entity.Replace($_.Name, [Regex]::Escape($_.Value))
+    switch ($entity.GetType().Name) {
+        'String' {
+            $params.GetEnumerator() | ForEach-Object {
+                if ($regexEscape -eq $false -or $null -eq $_.Value) {
+                    $entity = $entity.Replace($_.Name, $_.Value)
+                } else {
+                    $entity = $entity.Replace($_.Name, [Regex]::Escape($_.Value))
+                }
             }
+            break
+        }
+        'Object[]' {
+            $entity = $entity | ForEach-Object { substitute $_ $params $regexEscape }
+            break
+        }
+        'PSCustomObject' {
+            $entity = $entity.PSObject.Copy()
```

Comment:
Would not make sense to create local variable instead of "overriding" parameter?

Reply:
That's okay, too. I'll change it.

Reply:
Maybe this function should be rewrited, but in another PR.